vpr/src/base/read_options.cpp

```diff
@@ -2253,24 +2253,8 @@ void set_conditional_defaults(t_options& args) {
         args.anneal_sched_type.set(AUTO_SCHED, Provenance::INFERRED); // Otherwise use the automatic schedule
     }
 
-    //Are the pads free to be moved during placement or randomly locked to certain locations?
-    if (std::string(args.pad_loc_file) == "free") {
-        args.pad_loc_type.set(FREE, Provenance::INFERRED);
-
-        args.pad_loc_file.set("", Provenance::SPECIFIED);
-    } else if (std::string(args.pad_loc_file) == "random") {
-        args.pad_loc_type.set(RANDOM, Provenance::INFERRED);
-
-        args.pad_loc_file.set("", Provenance::SPECIFIED);
-    } else {
-        VPR_FATAL_ERROR(VPR_ERROR_UNKNOWN, "Unknown I/O pad location type\n");
-    }
-
     //Are the blocks locked to locations given by a constraints file?
-    if (std::string(args.constraints_file) == "not_locked") {
-        args.block_loc_type.set(NOT_LOCKED, Provenance::INFERRED);
-    } else {
-        args.block_loc_type.set(LOCKED, Provenance::INFERRED);
+    if (std::string(args.constraints_file) != "") {
         VTR_ASSERT(!args.constraints_file.value().empty());
```

Comment:
This assert doesn't look useful.  You are checking if the string is not empty, and then assert that it is not empty (the .empty() call boils down to the same check as the != "" call.  Delete this code block (if and assert) as it no longer does anything.